### DISCUSSION TITLE
Some versions of windows10 do not support WindowTransparencyLevel.Transparent

### DISCUSSION THREAD

**Discussion by 29654761** [AvaloniaUI/Avalonia#13253] [Q&A]
My software need a transparent background window. I used the property TransparencyLevelHint and set it to WindowTransparencyLevel.Transparent.  It works fine on most windows computers.

But not working on some very few versions of windows. The background is white not transparent. I checked ActualTransparencyLevel and it become WindowTransparencyLevel.None.

I don't know why the system can't support, or how to configurate it to support.  

I am also confused about whether TransparencyLevelHint should be used to create a transparent window?

**Comment by maxkatz6** [comment]
There always should be a fallback for systems, where it's not supported by Avalonia.
The proper way to support transparency on Windows is to use composition windowing APIs. 
On win8 it's direct composition APIs. And on win10 (from 10.0.15063 and newer, so not every win10 version) it's WinUI Composition APIs.
At this moment Avalonia supports only WinUI Composition APIs, but direct composition was planned, and some work has already been started.

Avalonia 0.10 had support for older (non composition API) approaches to make window transparent, but it was removed.

**Reply by 29654761** [reply]
Thinks, I'm looking forward to it.

**Reply by maxkatz6** [reply]
See https://github.com/AvaloniaUI/Avalonia/pull/13382
